src/transformers/utils/import_utils.py

```diff
@@ -747,13 +747,20 @@ def get_major_and_minor_from_version(full_version):
 
 @lru_cache
 def is_torch_xpu_available(check_device=False):
-    "Checks if `intel_extension_for_pytorch` is installed and potentially if a XPU is in the environment"
-    if not is_ipex_available():
+    """
+    Checks if XPU acceleration is available either via `intel_extension_for_pytorch` or
+    via stock PyTorch (>=2.4) and potentially if a XPU is in the environment
+    """
+    if not is_torch_available():
         return False
 
-    import intel_extension_for_pytorch  # noqa: F401
     import torch
 
+    if is_ipex_available():
+        import intel_extension_for_pytorch  # noqa: F401
+    elif not is_accelerate_available("0.33.0.dev"):
```

Comment:
@muellerzr : I added this check, since python version comparison evaluates `0.32.0.dev0 >= 0.32.0` as False, so I compared with `0.32.0.dev` and I am not sure that this is correct way. Please, advice.

Also, I can't raise exception here since `is_torch_xpu_available` is on a generic path and will fail non-xpu cases. And without exception the error which end user gets looks quite similar to what he will get running with wrong accelerate version. Do you have an idea where to raise exception notifying user that accelerate version is wrong?

Reply:
Ok, so, I remove accelerate check from `is_torch_xpu_available()` since I thought this function does its job and no need to add this here. And I added check with raising exception in here:
https://github.com/huggingface/transformers/blob/b7672826cad31e30319487af876e608d8af7d37b/src/transformers/training_args.py#L2138-L2140